test/hardhat.go

```diff
@@ -36,7 +38,17 @@ func (h *HardhatTest) NPMInstall() *HardhatTest {
 	return h
 }
 
+func (h *HardhatTest) SetEnv(key string, value string) *HardhatTest {
+	h.prevEnv[key] = os.Getenv(key)
+	err := os.Setenv(key, value)
+	if err != nil {
+		h.t.Fatalf("failed to set environment variable %s: %v", key, err)
+	}
+	return h
+}
```

Comment:
Added a function for setting environment variables before running the hardhat test. This is used to specify alternative TA files when running tests. If there was an existing env variable value, it is saved and restored later after the test is complete.

Reply:
@tahardi you should be able to use the test framework to set the environment rather than mutating the os environment with [t.SetEnv](https://pkg.go.dev/testing#T.Setenv)